extra/triggers.rst

```diff
@@ -0,0 +1,204 @@
+    $$ LANGUAGE plpgsql;
+  $cmd$);
+
+  -- Now create a trigger for every placement
+
+  SELECT run_command_on_placements(
+    'events',
+    $cmd$
+      CREATE TRIGGER events_set_author BEFORE INSERT OR UPDATE ON %s
+        FOR EACH ROW EXECUTE PROCEDURE set_author()
+    $cmd$
+  );
+
+Now if we try to add fake data, the author column will at least reveal who made the change:
+
+.. code-block:: postgresql
+
+  INSERT INTO events (description,author) VALUES ('a bad thing', 'wasnt-me');
+
+  TABLE events;
+
+::
+
+  ┌────┬─────────────┬────────┐
+  │ id │ description │ author │
+  ├────┼─────────────┼────────┤
+  │  1 │ a bad thing │ citus  │
+  └────┴─────────────┴────────┘
+
+The author says "citus" rather than "wasnt-me," showing this column can't be spoofed.
+
+**Trigger between colocated tables.**
+
+When two distributed tables are :ref:`colocated <colocation>`, then we can create a trigger to modify one based on changes in the other. The idea, once again, is to create triggers on the placements, but the trigger must be between pairs of placements that are themselves colocated. For this Citus has a special helper function ``run_command_on_colocated_placements``.
+
+Suppose that for every value inserted into ``little_vals`` we want to insert one twice as big into ``big_vals``.
+
+.. code-block:: postgresql
+
+  CREATE TABLE little_vals (key int, val int);
+  CREATE TABLE big_vals    (key int, val int);
+  SELECT create_distributed_table('little_vals', 'key');
+  SELECT create_distributed_table('big_vals',    'key');
+
+  -- This trigger function takes the destination placement as an argument
+
+  SELECT run_command_on_workers($cmd$
+    CREATE OR REPLACE FUNCTION embiggen() RETURNS TRIGGER AS $$
+      BEGIN
+        IF (TG_OP = 'INSERT') THEN
+          EXECUTE format(
+            'INSERT INTO %s (key, val) SELECT ($1).key, ($1).val*2;',
+            TG_ARGV[0]
+          ) USING NEW;
+        END IF;
+        RETURN NULL;
+      END;
+    $$ LANGUAGE plpgsql;
+  $cmd$);
+
+  -- Next we relate the co-located tables by the trigger function
+  -- on each co-located placement
+
+  SELECT run_command_on_colocated_placements(
+    'little_vals',
+    'big_vals',
+    $cmd$
+      CREATE TRIGGER after_insert AFTER INSERT ON %s
+        FOR EACH ROW EXECUTE PROCEDURE embiggen(%s)
+    $cmd$
+  );
+
+Then to test it:
+
+.. code-block:: postgresql
+
+  INSERT INTO little_vals VALUES (1, 42), (2, 101);
+  TABLE big_vals;
+
+::
+
+  ┌─────┬─────┐
+  │ key │ val │
+  ├─────┼─────┤
+  │   1 │  84 │
+  │   2 │ 202 │
+  └─────┴─────┘
+
+**Trigger between reference tables.**
+
+.. note::
+
+  **This workaround is only safe in limited situations.** When using such a trigger to insert into a reference table, make sure that no concurrent updates happen on the destination table. The order in which concurrent update/delete/insert commands are applied to replicas is not guaranteed, and replicas of the reference table can get out of sync with one another.
```

Comment:
To reinforce, I'd say that any DML to the destination then should be via the trigger only.